frontend-ssr/pages/submissions.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Head from 'next/head';
+import Footer from '~/components/footer/Footer.component';
+import Header from '~/components/header/Header.component';
+
+import styles from '../styles/pages/default-page.module.scss';
+import style from './submissions.module.scss';
+import ExSubmission from '../components/exercise-submissions/ExerciseSubmission';
+import ExerciseService from '../services/ExerciseSubmissions.service';
+import { loadSubmissions } from '~/redux/exercise-submissions/exercise-submissions.actions';
+import SubmissionSkeleton from '~/components/exercise-submissions/ExerciseSubmissionSkeleton';
+import SweetAlertService from '../services/sweet-alert/SweetAlert.service';
+import { Submission } from '~/redux/exercise-submissions/types';
+import SearchExerciseSubmissions from './searchExerciseSubmissions';
+
+interface State {
+  loading: boolean;
+}
+interface Props {
+  submissions: any;
+  dispatch: any;
+}
+
+class Submissions extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: false,
```

Comment:
I had a previous comment here but it seems nothing changed.

Loading should be based on whether or not we have submissions. In other words:

```
this.state = {
  loading: props.submissions.submissions === undefined
}
```